internal/ls/hover.go

```diff
@@ -131,149 +134,163 @@ func formatQuickInfo(quickInfo string) string {
+			}
+			declaration = core.Find(symbol.Declarations, ast.IsTypeAliasDeclaration)
+		default:
+			b.WriteString(c.TypeToStringEx(c.GetTypeOfSymbol(symbol), container, typeFormatFlags))
 		}
-		if flags&ast.SymbolFlagsInterface != 0 {
-			declaration = core.Find(symbol.Declarations, ast.IsInterfaceDeclaration)
-		}
-	case flags&ast.SymbolFlagsEnum != 0:
-		b.WriteString("enum ")
-		b.WriteString(c.SymbolToStringEx(symbol, container, ast.SymbolFlagsNone, symbolFormatFlags))
-	case flags&ast.SymbolFlagsModule != 0:
-		b.WriteString(core.IfElse(symbol.ValueDeclaration != nil && ast.IsSourceFile(symbol.ValueDeclaration), "module ", "namespace "))
-		b.WriteString(c.SymbolToStringEx(symbol, container, ast.SymbolFlagsNone, symbolFormatFlags))
-	case flags&ast.SymbolFlagsTypeParameter != 0:
-		b.WriteString("(type parameter) ")
-		tp := c.GetDeclaredTypeOfSymbol(symbol)
-		b.WriteString(c.SymbolToStringEx(symbol, container, ast.SymbolFlagsNone, symbolFormatFlags))
-		cons := c.GetConstraintOfTypeParameter(tp)
-		if cons != nil {
-			b.WriteString(" extends ")
-			b.WriteString(c.TypeToStringEx(cons, container, typeFormatFlags))
-		}
-		declaration = core.Find(symbol.Declarations, ast.IsTypeParameterDeclaration)
-	case flags&ast.SymbolFlagsTypeAlias != 0:
-		b.WriteString("type ")
-		b.WriteString(c.SymbolToStringEx(symbol, container, ast.SymbolFlagsNone, symbolFormatFlags))
-		writeTypeParams(&b, c, c.GetTypeAliasTypeParameters(symbol))
-		if len(symbol.Declarations) != 0 {
-			b.WriteString(" = ")
-			b.WriteString(c.TypeToStringEx(c.GetDeclaredTypeOfSymbol(symbol), container, typeFormatFlags|checker.TypeFormatFlagsInTypeAlias))
+		return declaration
+	}
+	var writeSymbol func(*ast.Symbol, bool) *ast.Node
+	writeSymbol = func(symbol *ast.Symbol, isAlias bool) *ast.Node {
+		var declaration *ast.Node
+		// Recursively write all meanings of alias
+		if symbol.Flags&ast.SymbolFlagsAlias != 0 {
+			if aliasedSymbol := c.GetAliasedSymbol(symbol); aliasedSymbol != c.GetUnknownSymbol() {
+				declaration = writeSymbol(aliasedSymbol, true /*isAlias*/)
+			}
 		}
-		declaration = core.Find(symbol.Declarations, ast.IsTypeAliasDeclaration)
-	case flags&ast.SymbolFlagsAlias != 0:
-		b.WriteString("import ")
-		b.WriteString(c.SymbolToStringEx(symbol, container, ast.SymbolFlagsNone, symbolFormatFlags))
-	default:
-		b.WriteString(c.TypeToStringEx(c.GetTypeOfSymbol(symbol), container, typeFormatFlags))
-	}
-	return b.String(), declaration
+		// Write the value meaning, if any
+		declaration = core.OrElse(declaration, writeSymbolMeaning(symbol, ast.SymbolFlagsValue|ast.SymbolFlagsSignature, isAlias))
+		// Write the type meaning, if any
+		declaration = core.OrElse(declaration, writeSymbolMeaning(symbol, ast.SymbolFlagsType&^ast.SymbolFlagsValue, isAlias))
+		// Write the namespace meaning, if any
+		declaration = core.OrElse(declaration, writeSymbolMeaning(symbol, ast.SymbolFlagsNamespace&^ast.SymbolFlagsValue, isAlias))
```

Comment:
Is unconditionally calling writeSymbolMeaning intentional? I think it is, but assumed this was a bug when I read it 😄